pkg/broker/ingress/args.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ingress
+
+import (
+	"context"
+
+	"cloud.google.com/go/pubsub"
+	cev2 "github.com/cloudevents/sdk-go/v2"
+	cepubsub "github.com/cloudevents/sdk-go/v2/protocol/pubsub"
+	"knative.dev/eventing/pkg/kncloudevents"
+)
+
+type Args struct {
+	IngressPort   int
+	ProjectID     string
+	PodName       string
+	ContainerName string
+}
+
+// NewHTTPMessageReceiver wraps kncloudevents.NewHttpMessageReceiver with type-safe options.
+func NewHTTPMessageReceiver(args Args) *kncloudevents.HttpMessageReceiver {
```

Comment:
I prefer only passing relevant args(the port number). Same for the NewPubSubClient below.